### PR TITLE
protocols/horizon/effects: Add UnmarshalJSON to SequenceBumped.

### PR DESCRIPTION
**Pull request by abuiles**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Add `UnmarshalJSON` to `SequenceBumped`,  this method supports `new_seq` as a `string` or a `int64` in the JSON payload. This will allow us to cut a new release for the horizon client which won't break once we change the data type on the JSON payload.

### Why

Using Int64 in JSON payloads produces inconsistent results in JS (see https://github.com/stellar/go/issues/1363).

We are in the process of updating all the int64 JSON fields to be of type string. The idea with this change is to extend the Unmarshal function to take either `int64` or `string`, that way we can release a version of the horizon client which won't break once the payload changes in the API.

See https://github.com/stellar/go/issues/1609

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: protocols/horizon/effects/main.go [hunk 1]
```diff
@@ -201,6 +202,42 @@ type SequenceBumped struct {
...
+
+	if s, ok := raw["new_seq"].(string); ok {
+		ns, err := strconv.ParseInt(s, 10, 64)
+		if err != nil {
+			return err
+		}
+
+		effect.NewSeq = ns
+	} else {
+		// letting Unmarshal handle the convertion so we don't have to convert
```
Comment (ire-and-curses) [inline protocols/horizon/effects/main.go]:
```suggestion
		// letting Unmarshal handle the conversion so we don't have to convert
```

File: clients/horizonclient/effect_request_test.go [hunk 2]
```diff
@@ -304,6 +304,46 @@ var firstEffectsPage = `{
...
+            "href": "https://horizon-testnet.stellar.org/effects?order=asc\u0026cursor=249108107265-1"
+          }
+        },
+        "id": "0000000249108107265-0000000001",
+        "paging_token": "249108107265-1",
+        "account": "GCQZP3IU7XU6EJ63JZXKCQOYT2RNXN3HB5CNHENNUEUHSMA4VUJJJSEN",
+        "type": "sequence_bumped",
+        "type_i": 43,
+        "created_at": "2019-06-03T16:36:24Z",
+        "new_seq": 300000000000
```
Comment (ire-and-curses) [inline clients/horizonclient/effect_request_test.go]: Let's add a comment somewhere or possibly split this into two tests, one for each unmarshalling. There's a lot of data here so it's not obvious that we're checking a string vs an int in the `new_seq` field

**Comment by tamirms** [comment]
you can use `json.Number` instead of an interface map:

https://play.golang.org/p/epDKjGKeg7d

**Comment by abuiles** [comment]
@ire-and-curses I split the test so it's clear what is being tested.

@tamirms I  updated the code to use `json.Number` -  thanks!

**Pull request merged**